espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/EspressoW3CActionAdapter.java

```diff
@@ -489,4 +493,25 @@ public void sleep(long duration) throws AppiumException {
     public Logger getLogger() {
         return AndroidLogger.logger;
     }
+
+    // If a 'mouse' event was provided, but it's a touchscreen, switch to 'touch'
+    // This is because some clients only send 'mouse' events and the assumption is that if they
+    // send 'mouse' events to a device that has a touch screen, it needs to be converted
+    // This may need to be re-visited in the future if we wish to support Android laptops
```

Comment:
Android laptops usually have touch screen as well ;)

Reply:
I know. That's why we may need to revisit this in the future. It might be a problem for some to always have mouse actions converted to touch.

Reply:
which clients only support mouse? can we just send a simple PR to fix them instead?

Reply:
it is not about they are only support mouse, but the default value is `mouse` there. And it is not so easy to change that